### PR TITLE
Add new gang/locus type to support functions like pg_locks_status.

### PR DESCRIPTION
**Pull request by pengzhout**
Internal functions such as  pg_locks_status() and gp_stat_activity() return
sets of tuples collected both from master and segments. Current implement need
functions to do their own dispatch on master, but in most queries, a entry db
is assigned to execute the function scan which make the wrong results because
GPDB do not allow the QE do their own dispatching.

This fix added a new locus called CdbLocusType_Mixed which indicate data is
partitioned on both master and segments with unknown distributed column. It also
added a new gang type named GANGTYPE_PRIMARY_MIXED which let dispatcher to assign
a entry db gang and a N-segments gang working for a single slice. In this way,
functions can be dispatched to all segment and master, and return the right results.

Broadcast and gather motion works fine no mater it is 1 -> N+1, N -> N+1 ,
N+1 -> N or N+1 ->1. Redistributed motion N -> N+1 is not support now, in fact
you can not construct a N -> N+1 redistributed motion except union operation because
CdbLocusType_Mixed has no distributed key. For union operation,N -> N+1 redistribution
is disabled.

### CODE REVIEW

**Comment by entong** [comment]
Hi Pengzhou, do you need the Orca fallback story to be completed to get a green build with Orca on?

**Comment by pengzhout** [comment]
Yes, i will hold the merging until orca fallback story is completed. Here,
i hope dev guys can review the code firstly and gather more comments.

**Comment by hlinnaka** [comment]
There's a fair amount whitespace differences etc. cleanup included in this. I like cleaning up, but would be good to split those into a separate commit, to make it more clear what the actual changes.

**Comment by atris** [comment]
I will be working on fallback story next week. It will be a bit slow since I am still taking baby steps in Orca. I hope that is fine.

**Comment by ivannovick** [comment]
Thx @atris

File: src/backend/cdb/motion/ic_common.c [hunk 1]
```diff
@@ -1017,7 +1017,12 @@ void adjustMasterRouting(Slice *recvSlice)
 		if (cdbProc)
 		{
 			if (cdbProc->listenerAddr == NULL)
-				cdbProc->listenerAddr = pstrdup(MyProcPort->remote_host);
+			{
+				if (!strcmp(MyProcPort->remote_host, "[local]"))
+					cdbProc->listenerAddr = pstrdup("127.0.0.1");
+				else
```
Comment (yydzero) [inline src/backend/cdb/motion/ic_common.c]: Why need to handle '[local]' specially?

If hardcode 127.0.0.1, is it possible to break IPV6 env?

File: src/backend/cdb/cdbdisp.c [hunk 2]
```diff
@@ -143,7 +143,7 @@ addSegDBToDispatchThreadPool(DispatchCommandParms  *ParmsAr,
 							 CdbDispatchResult     *dispatchResult);
 
 static void
-cdbdisp_dispatchCommandToAllGangs(const char	*strCommand,
+cdbdisp_dispatchSetCommandToAllGangs(const char	*strCommand,
```
Comment (yydzero) [inline src/backend/cdb/cdbdisp.c]: Would be great to add more comments for functions.

File: src/backend/cdb/motion/ic_common.c (hunk 1 above)
Comment (pengzhout) [inline src/backend/cdb/motion/ic_common.c]: +1 that may break the ipv6.

**Comment by pengzhout** [comment]
Closed, will create a new one after orca modification included.

**Pull request closed without merging**